daemon/api.go

```diff
@@ -1481,6 +1487,36 @@ func setSnapConf(c *Command, r *http.Request, user *auth.UserState) Response {
 	return AsyncResponse(nil, &Meta{Change: change.ID()})
 }
 
+func getSnapHistory(c *Command, r *http.Request, user *auth.UserState) Response {
+	vars := muxVars(r)
+	name := vars["name"]
+
+	route := c.d.router.Get(snapCmd.Path)
+	if route == nil {
+		return InternalError("cannot find route for snaps")
+	}
+
+	url, err := route.URL("name", name)
+	if err != nil {
+		return InternalError("cannot build URL for %q snap: %v", name, err)
+	}
+
+	var history []map[string]interface{}
+	snapRevisions, err := allLocalRevisions(c.d.overlord.State(), name)
+	if err != nil {
+		if err == state.ErrNoState {
+			return NotFound(err.Error())
+		}
+		return InternalError(err.Error())
```

Comment:
add some context here to explain what you were trying to do when things went wrong (see the above `InternalError`s for example).